terraform/instances.tf

```diff
@@ -100,7 +100,13 @@ resource "aws_instance" "api" {
   }
 
   provisioner "file" {
-    content     = data.template_file.sup_service.rendered
+    content = templatefile(
```

Comment:
I'm not certain we can use `templatefile` here. When we were looking at the issue with Windows workers not getting the peer list, we looked at this function and it read like it wouldn't receive any changes made as part of a terraform apply operation. 


> Functions do not participate in the dependency graph, so this function cannot be used with files that are generated dynamically during a Terraform operation.

https://www.terraform.io/docs/configuration/functions/templatefile.html

Reply:
I might be mistaken, but I believe if we assign the `templatefile` output to a local variable it will then be injected in the dep graph. It would also DRY up the `file` provisioner content block duplication:

```
provider "aws" {
  region  = var.aws_region
  profile = "habitat"
}

locals {
  hab_sup_service_config = templatefile(
      "${path.module}/templates/hab-sup.service.tpl",
      {
        flags = "--auto-update --peer ${join(" ", var.peers)} --channel ${var.sup_release_channel} --listen-gossip 0.0.0.0:${var.gossip_listen_port} --listen-http 0.0.0.0:${var.http_listen_port}"
        log_level = var.log_level
        enabled_features = var.enabled_features
      })
}
...
  provisioner "file" {
    content = local.hab_sup_service_config
    destination = "/home/ubuntu/hab-sup.service"
  }
```

Reply:
Yeah the wording is a bit vague - the template file we are pointing to is always on disk, and is not dynamically generated so we should be ok?

Reply:
@chefsalim That's how I was interpreting it, too.

@jeremymv2 Ah, that's what I was looking for... I'll give that a shot. The duplication was the part of this PR I liked the least.

Reply:
local variable usage in https://github.com/habitat-sh/builder/pull/1252/commits/8ce71df0fc3aefc85cc236d48c3cfebae4d73701